iceoryx_examples/icehello/iox_subscriber_helloworld.cpp

```diff
@@ -14,50 +14,58 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
+#include "topic_data.hpp"
+
 #include "iceoryx_posh/popo/subscriber.hpp"
 #include "iceoryx_posh/runtime/posh_runtime.hpp"
 #include "iceoryx_utils/posix_wrapper/signal_handler.hpp"
-#include "topic_data.hpp"
 
-#include <chrono>
 #include <iostream>
 
 bool killswitch = false;
+constexpr char APP_NAME[] = "iox-cpp-subscriber-helloworld";
 
-static void sigHandler(int sig [[gnu::unused]])
+static void sigHandler(int f_sig [[gnu::unused]])
 {
+    // caught SIGINT or SIGTERM, now exit gracefully
     killswitch = true;
 }
 
-void receive()
+int main()
 {
-    iox::popo::SubscriberOptions subscriberOptions;
-    subscriberOptions.nodeName = "uMgungundlovu";
+    // register sigHandler
+    auto signalIntGuard = iox::posix::registerSignalHandler(iox::posix::Signal::INT, sigHandler);
+    auto signalTermGuard = iox::posix::registerSignalHandler(iox::posix::Signal::TERM, sigHandler);
 
-    iox::popo::Subscriber<CounterTopic> subscriber({"Group", "Instance", "Counter"}, subscriberOptions);
+    // initialize runtime
+    iox::runtime::PoshRuntime::initRuntime(APP_NAME);
 
+    // initialized subscriber
+    iox::popo::Subscriber<RadarObject> subscriber({"Radar", "FrontLeft", "Object"});
+
+    // run until interrupted by Ctrl-C
     while (!killswitch)
     {
-        std::this_thread::sleep_for(std::chrono::seconds(1));
+        auto takeResult = subscriber.take();
 
-        while (subscriber.hasData())
+        if (!takeResult.has_error())
         {
-            subscriber.take()
-                .and_then([](auto& sample) { std::cout << "Received: " << *sample.get() << std::endl; })
-                .or_else([](auto&) { std::cout << "Error while receiving." << std::endl; });
-        };
-        std::cout << "Waiting for data ... " << std::endl;
-    }
-}
-
-int main()
-{
-    auto signalIntGuard = iox::posix::registerSignalHandler(iox::posix::Signal::INT, sigHandler);
-    auto signalTermGuard = iox::posix::registerSignalHandler(iox::posix::Signal::TERM, sigHandler);
-    iox::runtime::PoshRuntime::initRuntime("iox-subscriber");
+            std::cout << APP_NAME << " got value: " << takeResult.value()->x << std::endl;
+        }
+        else
+        {
+            if (takeResult.get_error() == iox::popo::ChunkReceiveResult::NO_CHUNK_AVAILABLE)
```

Comment:
Aren't we here flooded with "No chunk available" messages since the subscriber is 10 times faster than the publisher?

Reply:
I quite liked this ouput when running icehello and iceensemble. The users learns here that this approach is a polling one. And not the best approach ;)